File: trunk/src/platform_iphone.m
//
//  platform_iphone.m
//
//  Created by Matthias Ringwald on 8/15/09.
//

#import "platform_iphone.h"

#ifdef USE_SPRINGBOARD

#ifdef __OBJC__
#import <Foundation/Foundation.h>
#import <UIKit/UIKit.h>
#endif

@interface UIApplication (privateStatusBarIconAPI)
- (void)addStatusBarImageNamed:(id)fp8 removeOnAbnormalExit:(BOOL)fp12;
- (void)addStatusBarImageNamed:(id)fp8;
- (void)removeStatusBarImageNamed:(id)fp8;
// iPhoneOS >= 2.2
- (void)addStatusBarImageNamed:(id)fp8 removeOnExit:(BOOL)fp12;
@end

// update SpringBoard icon
void platform_iphone_status_handler(BLUETOOTH_STATE state){
    printf("iphone state %u\n", state);
	UIApplication *theApp = [UIApplication sharedApplication];
    printf("the app %u\n", (int) theApp);
    [theApp addStatusBarImageNamed:@"BluetoothActive"];
    
    switch (state) {
        case BLUETOOTH_OFF:
            break;
        case BLUETOOTH_ON:
            break;
        case BLUETOOTH_ACTIVE:
            break;
        default:
            break;
    }
}

#endif

File: trunk/src/platform_iphone.c
//
//  platform_iphone.c
//
//  Created by Matthias Ringwald on 8/15/09.
//

#include "platform_iphone.h"

#include "../config.h"
#include "../SpringBoardAccess/SpringBoardAccess.h"

#include <stdio.h>

#ifdef USE_SPRINGBOARD

// update SpringBoard icons
void platform_iphone_status_handler(BLUETOOTH_STATE state){
    switch (state) {
        case BLUETOOTH_OFF:
            SBA_removeStatusBarImage("BTstack");
            SBA_removeStatusBarImage("BTstackActive");
            printf("Bluetooth status: OFF\n");
            break;
        case BLUETOOTH_ON:
            SBA_removeStatusBarImage("BTstackActive");
            SBA_addStatusBarImage("BTstack");
            printf("Bluetooth status: ON\n");
            break;
        case BLUETOOTH_ACTIVE:
            SBA_removeStatusBarImage("BTstack");
            SBA_addStatusBarImage("BTstackActive");
            printf("Bluetooth status: ACTIVE\n");
            break;
        default:
            break;
    }
}

#endif